pandas/core/indexers.py

```diff
@@ -152,6 +152,8 @@ def check_setitem_lengths(indexer, value, values) -> bool:
         if is_list_like(value):
             if len(indexer) != len(value):
                 # boolean with truth values == len of the value is ok too
+                if isinstance(indexer, list):
```

Comment:
should't this be is_list_like? (e.g. if the indexer is a np.array?)

Reply:
If we already have an array, we don't have to cast to array